nmostesting/suites/IS1101Test.py

```diff
@@ -2056,26 +2130,32 @@ def test_04_04_01(self, test):
         The test requires streaming from a Sender
         in order to verify the state of the Receiver.
         """
+        if len(self.receivers) == 0:
+            return test.UNCLEAR("No IS-11 receivers")
         if len(CONFIG.IS11_REFERENCE_SENDER) == 0:
-            return test.FAIL("IS-11 refecence sender is not configured")
+            return test.DISABLED("Please configure IS11_REFERENCE_SENDER in Config.py")
         if len(self.senders) == 0:
             return test.DISABLED("No IS-11 reference sender")
         for sender in self.senders:
-            url = self.conn_url.replace(PATTERN.search(self.conn_url)[0], CONFIG.IS11_REFERENCE_SENDER)
+            url = self.conn_url.replace(urlparse(self.conn_url).hostname, CONFIG.IS11_REFERENCE_SENDER)
             json_data = {
                     "master_enable": True,
                     "activation": {"mode": "activate_immediate"}
                 }
             headers = {"Content-Type": "application/json"}
-            _, response = TestHelper.do_request('PATCH', url + "single/senders/"
-                                                + sender["id"] + "/staged/", json=json_data, headers=headers)
-            time.sleep(3)
+            valid_patch, response = TestHelper.do_request('PATCH', url + "single/senders/"
+                                                          + sender["id"] + "/staged/", json=json_data, headers=headers)
+            if not valid_patch:
+                return test.FAIL("Unexpected response from the connection API: {}".format(response))
+            time.sleep(CONFIG.HTTP_TIMEOUT*3)
```

Comment:
Why sleep here? Would it at least be after the `if response.status_code != 200:` block, i.e. _before_ the next request rather than _after_ this one?